apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/DLQGrpcRdaSource.java

```diff
@@ -0,0 +1,275 @@
+        Preconditions.checkNotNull(claimType),
+        callOptionsFactory,
+        appMetrics);
+    this.dao = new DLQDao(Preconditions.checkNotNull(entityManager));
+    this.sequencePredicate = sequencePredicate;
+  }
+
+  /**
+   * Checks for {@link MessageError}s in the database, then calls the service through the specific
+   * implementation of GrpcStreamCaller provided to our constructor. Cancels the response stream if
+   * reading from the stream is interrupted.
+   *
+   * @param maxPerBatch maximum number of objects to collect into a batch before calling the sink
+   * @param sink to receive batches of objects
+   * @return the number of objects that were successfully processed
+   * @throws ProcessingException wrapper around any Exception thrown by the service or sink
+   */
+  @Override
+  public int retrieveAndProcessObjects(int maxPerBatch, RdaSink<TMessage, TClaim> sink)
+      throws ProcessingException {
+    int totalProcessed = 0;
+
+    MessageError.ClaimType type = MessageError.ClaimType.valueOf(claimType.toUpperCase());
+
+    List<MessageError> messageErrors =
+        dao.findAllMessageErrorsByClaimTypeAndStatus(type, MessageError.Status.UNRESOLVED);
+
+    final Set<Long> sequenceNumbers =
+        messageErrors.stream().map(MessageError::getSequenceNumber).collect(Collectors.toSet());
+
+    if (sequenceNumbers.isEmpty()) {
+      log.info("Found no {} claims in DLQ, skipping", claimType);
+    } else {
+      log.info(
+          "Found {} {} claims in DLQ, attempting to reprocess", sequenceNumbers.size(), claimType);
+
+      totalProcessed =
+          tryRetrieveAndProcessObjects(dlqProcessingLogic(sink, type, sequenceNumbers));
+    }
+
+    return totalProcessed;
+  }
+
+  /**
+   * Helper method, called internally, to return the DLQ processing logic to pass to {@link
+   * AbstractGrpcRdaSource#tryRetrieveAndProcessObjects(Processor)}.
+   *
+   * @param sink The sink to use to write claims, etc.
+   * @param type The {@link MessageError.ClaimType} associated with the claims to process.
+   * @param sequenceNumbers A {@link Set} of sequence numbers to attempt to reprocess
+   * @return The DLQ processing logic {@link Processor}.
+   */
+  @VisibleForTesting
+  Processor dlqProcessingLogic(
+      RdaSink<TMessage, TClaim> sink, MessageError.ClaimType type, Set<Long> sequenceNumbers) {
+    return () -> {
+      ProcessResult processResult = new ProcessResult();
+      final String apiVersion = caller.callVersionService(channel, callOptionsFactory.get());
+
+      for (final long startingSequenceNumber : sequenceNumbers) {
+        log.info(
+            "calling API for {} claims starting at sequence number {}",
+            claimType,
+            startingSequenceNumber);
+
+        // The "since" parameter of the RDA API is actually non-inclusive, so we have to subtract 1
+        // in order to attempt to get the claim we really want
+        final long SINCE = startingSequenceNumber - 1;
```

Comment:
It is a workaround, we don't have control over the language the upstream RDA API uses or how it defines it.  We define our configuration as "Where do you want to start", but the RDA API's definition for "since" is to start at the next value after that, so we have to subtract one.  The goal is to keep the own configuration (and associated language) intuitive.